core/wave-autoscale/src/scaling_component/netfunnel_segment.rs

```diff
@@ -0,0 +1,152 @@
+use super::ScalingComponent;
+use anyhow::{Ok, Result};
+use async_trait::async_trait;
+
+use data_layer::ScalingComponentDefinition;
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::collections::HashMap;
+
+use reqwest::Client;
+
+pub struct NetfunnelSegmentScalingComponent {
+    definition: ScalingComponentDefinition,
+}
+
+impl NetfunnelSegmentScalingComponent {
+    // Static variables
+    pub const SCALING_KIND: &'static str = "netfunnel";
+
+    // Functions
+    pub fn new(definition: ScalingComponentDefinition) -> Self {
+        NetfunnelSegmentScalingComponent { definition }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NetfunnelResponse {
+    #[serde(rename = "statusCode")]
+    status_code: i32,
+
+    #[serde(rename = "status")]
+    status: String,
+
+    #[serde(rename = "errorCode")]
+    error_code: String,
+
+    #[serde(rename = "message")]
+    message: String,
+}
+
+#[async_trait]
+impl ScalingComponent for NetfunnelSegmentScalingComponent {
+    fn get_scaling_component_kind(&self) -> &str {
+        &self.definition.component_kind
+    }
+    fn get_id(&self) -> &str {
+        &self.definition.id
+    }
+    async fn apply(&self, params: HashMap<String, serde_json::Value>) -> Result<()> {
+        let metadata: HashMap<String, serde_json::Value> = self.definition.metadata.clone();
+
+        if let (
+            Some(serde_json::Value::String(authorization)),
+            Some(serde_json::Value::String(organization_id)),
+            Some(serde_json::Value::String(tenant_id)),
+            Some(serde_json::Value::String(user_key)),
+            Some(serde_json::Value::String(project_id)),
+            Some(serde_json::Value::String(segment_id)),
+            max_inflow,
+        ) = (
+            metadata.get("authorization"),
+            metadata.get("organization_id"),
+            metadata.get("tenant_id"),
+            metadata.get("user_key"),
+            metadata.get("project_id"),
+            metadata.get("segment_id"),
+            params
+                .get("max_inflow")
+                .and_then(serde_json::Value::as_i64)
+                .map(|v| v as i32),
+        ) {
+            if let Some(max_inflow) = max_inflow {
+                let url = format!(
+                    "https://dev-api.surffy-dev.io/v2/wave/project/{}/segment/{}",
```

Comment:
Please obtain the base url (e.g. https://dev-api.surffy-dev.io) from the metadata